pkg/discovery/worker/aggregation/container_utilization_data_aggregator.go

```diff
@@ -37,24 +38,33 @@ func (allDataAggregator *allUtilizationDataAggregator) String() string {
 	return allDataAggregator.aggregationStrategy
 }
 
-func (allDataAggregator *allUtilizationDataAggregator) Aggregate(commodities []*proto.CommodityDTO) ([]float64, error) {
-	if len(commodities) == 0 {
-		err := fmt.Errorf("error to aggregate commodities using %s : commodities list is empty", allDataAggregator)
-		return []float64{}, err
+func (allDataAggregator *allUtilizationDataAggregator) Aggregate(resourceMetrics *repository.ContainerMetrics) ([]float64, int64, int32, error) {
+	if len(resourceMetrics.Used) == 0 {
+		err := fmt.Errorf("error aggregating container utilization data using %s: used data points list is empty", allDataAggregator)
+		return []float64{}, 0, 0, err
+	}
+	capacity := resourceMetrics.Capacity
+	if capacity == 0.0 {
+		err := fmt.Errorf("error aggregating container utilization data using %s: capacity is 0", allDataAggregator)
+		return []float64{}, 0, 0, err
 	}
 	var utilizationDataPoints []float64
-	for _, commodity := range commodities {
-		used := *commodity.Used
-		capacity := *commodity.Capacity
-		if capacity == 0.0 {
-			err := fmt.Errorf("error to aggregate %s commodities using %s : capacity is 0", commodity.CommodityType,
-				allDataAggregator)
-			return []float64{}, err
-		}
-		utilization := used / capacity * 100
+	var firstTimestamp int64
+	var lastTimestamp int64
+	for _, usedPoint := range resourceMetrics.Used {
+		utilization := usedPoint.Value / capacity * 100
 		utilizationDataPoints = append(utilizationDataPoints, utilization)
+		if firstTimestamp == 0 || usedPoint.Timestamp < firstTimestamp {
+			firstTimestamp = usedPoint.Timestamp
+		}
+		lastTimestamp = int64(math.Max(float64(lastTimestamp), float64(usedPoint.Timestamp)))
```

Comment:
I think it's really not a good idea to cast to float64 first because these timestamps are pretty large numbers and so if they are close together the loss in precision can actually have a significant impact on the calculation.

It's unfortunate there's no nice way to take the max of two integer types in go, but it would be better to use regular comparison operators here instead of casting to float64 and back.

Reply:
@marvelyue @dblinn What would be your comment about keeping this as the golangs origin type `Time.time`? The comparisons could safely be done, using the comparisons available in `Time` package, on that then. In case server does not agree with it, we can cast the final value as the server accepted type after all this processing is complete.

Reply:
Thanks @irfanurrehman for the suggestion. I just updated the code by simply using normal comparison on int64 here per @dblinn's suggestion:
```
                if usedPoint.Timestamp > lastTimestamp {
			lastTimestamp = usedPoint.Timestamp
		}
```
I think using the comparison in `Time` package (like `After` `Before`) would be similar as comparing int64 values above. Either should work